docs/coding-conventions.md

```diff
@@ -0,0 +1,89 @@
+# Coding conventions
+
+This document outlines a collection of guidelines, style suggestions, and tips
+for writing code in the KubeVirt project. It is partially based on [Kubernetes
+Coding Conventions](https://github.com/kubernetes/community/blob/master/contributors/guide/coding-conventions.md).
+
+- [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+- [Effective Go](https://golang.org/doc/effective_go.html)
+- Know and
+  avoid [Go landmines](https://gist.github.com/lavalamp/4bd23295a9f32706a48f)
+- Inline variable assignments when possible
+    - For example inline `err` checks
+- Use early returns to avoid unnecessarily deeply nested if/else blocks
+- Use switch-cases to avoid long if/else if/else blocks
+- Avoid named return values
```

Comment:
I strongly disagree.
Let's continue here https://github.com/kubevirt/kubevirt/issues/11446

Reply:
I agree, we should not forbid them completely but make authors think twice before using them. Can we gather a list of good reasons to use them here?